pdns/auth-main.hh

```diff
@@ -48,7 +48,7 @@ extern double avg_latency;
 extern std::unique_ptr<TCPNameserver> TN;
 extern void declareArguments();
 extern void declareStats();
-extern void mainthread();
+void mainthread();
```

Comment:
looks like this line can go entirely

Reply:
.. in which case `mainthread` in `auth-main.cc` can even be declared `static`

Reply:
`extern` for function prototypes is very old-school anyway

Reply:
Right, I was starting to suspect a bigger cleanup is in order here. I'll merge this one because it quiets the compiler, and then we can do better later.